spec/models/account_filter_spec.rb

```diff
@@ -18,4 +18,29 @@
       expect { filter.results }.to raise_error(/wrong/)
     end
   end
+
+  describe 'with origin and by_domain interacting' do
+    let!(:local_account) { Fabricate(:account, domain: nil) }
+    let!(:remote_account) { Fabricate(:account, domain: 'example.org') }
```

Comment:
You need a second remote account with another domain to trigger test failures for `works with domain first and origin remote`

Reply:
Thanks -- will add that to specs and push an update.

Reply:
One could argue this changes the behavior when `origin=remote` and `by_domain` is set to something: one would override the other while the old behavior was to return an empty list. But this probably does not matter much as that query was nonsensical in the first place.

Reply:
Yeah -- seems relatively low risk to change what was unspecified behavior before.